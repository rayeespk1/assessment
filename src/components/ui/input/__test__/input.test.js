import React from "react";
import '@testing-library/jest-dom'
import { render, cleanup } from "@testing-library/react";
import Input from "../input";

afterEach(cleanup);

test("renders", () => {
    const elementConfig = {
        type: 'text',
        placeholder: 'Item Name'
    };
    const validation = {
        required: true
    }
    const { asFragment } = render(<Input
        elementType="input"
        elementConfig={elementConfig}
        value=""
        invalid={true}
        shouldValidate={validation}
        touched={false}
        changed={() => null} />);
    expect(asFragment()).toMatchSnapshot();
});