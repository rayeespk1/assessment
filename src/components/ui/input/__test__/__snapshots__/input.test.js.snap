// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="Input"
  >
    <input
      class="InputElement"
      placeholder="Item Name"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
